### ISSUE TITLE
Dual E5-2690 with RAMDisk is slow compare to others.

### ISSUE THREAD

**Issue by zeroarst** [madMAx43v3r/chia-plotter#41]
I have a server which has dual E5-2690 I think it is the one: https://ark.intel.com/content/www/us/en/ark/products/64596/intel-xeon-processor-e5-2690-20m-cache-2-90-ghz-8-00-gt-s-intel-qpi.html
Physical cores: 8x2, threads: 16x2
It has 128 DDR3 RAM, I created a RAMDisk with ImDisk Toolkit. Allocated 110GB for it. Path is /mnt/r/
Also two 2T M.2 NVME SSD. /mnt/d/ and /mnt/e but I only used one for this test.

I used WSL2 and the result was as below. I didn't even finish it as I think it was slow compare to others so I stopped it. Tried both 16 and 32 threads, not much different it seemed.
```
Number of Threads: 32
Number of Sort Buckets: 2^7 (128)
Pool Public Key:   af2172f8be6a13fe560bdcc6b4a64e12459deedf12526901410ab7e97de3a3f1e119d3cff9bb7aabbfd733fa0fa2ae07
Farmer Public Key: 902dde7cc1501d1df4d8d689f75e36e6f912c8b61edbd570cb901024687abdfd13b0bae5f57d672e6b2339d079c7cab6
Working Directory:   /mnt/d/
Working Directory 2: /mnt/r/
Plot Name: plot-k32-2021-06-08-21-45-5e789c389f87f9e23df0c4a2af0b1e3558d369e55f3e2e162f4e12963c591b46
[P1] Table 1 took 170.16 sec
[P1] Table 2 took 803.987 sec, found 4294975885 matches
[P1] Lost 8503 matches due to 32-bit overflow.
[P1] Table 3 took 945.028 sec, found 4295069307 matches
```

I compared to my big brother's PC, which is 5800x. He dose not even use RAMDisk, however, still faster than me. 
![zzero_chia_plotter_full_cpu_usage jpg](https://user-images.githubusercontent.com/1834837/121200951-7a381c00-c8b7-11eb-9d6c-fbb144307d9a.jpeg)

When his PC was plotting at phase 1, its CPU has nearly 100% usage... 
![zzero_chia_plotter_full_cpu_usage](https://user-images.githubusercontent.com/1834837/121201608-ff233580-c8b7-11eb-943a-70d3fe9048b4.jpg)

Mine did hit 100% with 32 threads and even 50%. I didn't take the screenshot, might test again and do it tomorrow.

I also got another PC, which has the same problem. Somehow the CPU usage is just not 100%.
![roy_new_pc_chia_plot](https://user-images.githubusercontent.com/1834837/121202295-883a6c80-c8b8-11eb-857d-b9a4d12d84a0.jpeg)

Is it normal? Did I miss anything..?

**Comment by VertiHydro** [comment]
DDR3 can't be helping, but WSL2 might have something to do with it as well. Have you tried compiling natively for Windows (might be a bit difficult).

**Comment by zeroarst** [comment]
I did. With WSL2. Maybe I will try to compile again tomorrow..
Are you saying DDR3 is the problem?

**Comment by GTANAdam** [comment]
DDR3 can't be the problem, it can handle 10667 MB/s @ 1333, it seems like you have a bottleneck. why not use Linux?

**Comment by osxme** [comment]
I think you should use ubuntu

**Comment by VertiHydro** [comment]
> I compiled in WSL2, was not too difficult. Maybe I will try to compile again tomorrow..
> Are you saying DDR3 is the problem?

Yes sorry I meant instead of compiling for Linux on WSL2, compile natively for windows. This can be done with something like minGW, but as I can now see in #38, that might be a bit too hard. Have you considered trying to boot Ubuntu from a USB stick? That might at least demonstrate whether or not WSL2 is to blame.

**Comment by Smallthing** [comment]
this make me laugh
/mnt/d/
if it like my thought,your use wsl2's automount windows drive d: which is a ntfs drive or so and io with something like network.so the io though is very low.
can you plot inside the wsl2 with ext4 or xfs,(and wsl2 even can mount ext4 xfs raw disk now)

**Comment by Smallthing** [comment]
sorry for chinese because i have something cant metion in my english
让我忍不住笑是因为你不知道wsl2和win侧的io对接的问题。。。

**Comment by VertiHydro** [comment]
Ohh haha I actually had no idea about /mnt and the network share... that is hilarious! So, future note for anybody wanting to plot on Windows: do NOT try and plot on a drive that is automatically mounted at /mnt/. The performance will be horrific. Instead, you must either stop using windows altogether, compile the program natively for Windows, (as I said above, not easy), or create a linux filesystem within WSL2. This issue is mentioned here: https://github.com/microsoft/WSL/issues/873

**Comment by zeroarst** [comment]
> sorry for chinese because i have something cant metion in my english
> 让我忍不住笑是因为你不知道wsl2和win侧的io对接的问题。。。却可以编译出来。。。

哈 我確實不知道額，業餘的編程人士啊，見笑了。我會試試ext4或xfs

**Comment by PeratX** [comment]
Try WSL1 for better filesystem performance
WSL1对于跨文件系统更加友好

**Comment by VertiHydro** [comment]
Noooo don't do that - just use linux - please it's not worth it - WSL1 has a lot of problems - large chance you will run into some compilation issue. Just try booting Ubuntu from a USB.

**Comment by zeroarst** [comment]
Dose that mean I should also create RAMDisk inside WSL2?

**Comment by Smallthing** [comment]
> Dose that mean I should also create RAMDisk inside WSL2?

YES

**Comment by PeratX** [comment]
> Noooo don't do that - just use linux - please it's not worth it - WSL1 has a lot of problems - large chance you will run into some compilation issue. Just try booting Ubuntu from a USB.

I mean WSL1 is just a better environment to test it. instead of using it in production.

**Comment by merakliseyit** [comment]
I have a similar server. dl380 gen9

my server is 2x 2697A v4 + 384gb ddr 4 2133 mhz 4x 2tb nvme
I'm running madmax on windows. (I don't know anything about linux, ubuntu) phase 1 duration is 2400 seconds.this is very bad
I tried many alternatives.
tmp1=nvme / tmp2=nvme
tmp1=nvme / tmp2= ramdisk
tmp1=ramdisk / tmp2=ramdisk
There is no difference. even when i use ramdisk the time is extended by 4 minutes. what is the reason of this? I've seen it finish in 1800 seconds with servers much older than my server.where is the error please help

**Comment by leonidbkh** [comment]
Your brother's 5800x is not working on full speed, i have better results. I think its cause of slow nvme.
